src/server/routes/adminPanel.tsx

```diff
@@ -26,29 +26,35 @@ import {snakeCase as _snakeCase, isNil} from 'lodash';
 import {escapeProps, generateProps} from '../helpers/props';
 import AdminPanelSearchPage from '../../client/components/pages/admin-panel-search';
 import Layout from '../../client/containers/layout';
-import {PrivilegeTypes} from '../../common/helpers/privileges-utils';
+import {PrivilegeType} from '../../common/helpers/privileges-utils';
 import React from 'react';
 import ReactDOMServer from 'react-dom/server';
 import express from 'express';
 import target from '../templates/target';
 
 
-const ADMIN = PrivilegeTypes.ADMIN_PRIV.value;
+type PassportRequest = express.Request & {
+	user: any,
+	session: any,
+	query: any
+};
```

Comment:
I'm not a fan of having to redeclare this type in every module which uses custom request properties. This will only get worse once we upgrade more routes from JS to TS.
I know that there is already existing code which does this, but even having this declaration in one place and importing it to overwrite the standard parameter type of so many request handlers is suboptimal.

Fortunately, there is a better way to declare these properties in one place which requires no change at all in all of these route handler modules:
```ts
// Extend the express Request type via declaration merging
// Precise type declarations for `user` and `session` would be even better, of course.
declare module 'express-serve-static-core' {
	interface Request {
		user: any;
		session: any;
	}
}
```
It does not really matter in which module this is declared, but ideally it should reside wherever these properties are attached to the standard request. This would probably be in `auth.ts` and `session.js` (then `session.ts`), i.e. in two steps of declaration merging.

Overwriting the `query` property with `any` should not be done in any case, by the way. It may mute the type error because the default query parser has an ugly return type of `string | ParsedQs | string[] | ParsedQs[]` for each query parameter while we usually expect a string, but this is not safe.
When I was working on the Wikipedia PR, I had a similar problem and used a workaround to avoid having to use express' query parser:

https://github.com/metabrainz/bookbrainz-site/blob/1029d42f3f710cc3e1aeba525ead1b443d131d40/src/server/routes/wikimedia.ts#L25-L36

Maybe we should move that helper method into our utilities and make it the new standard? :grin:

Reply:
Thanks for these suggestions. I've made some changes.